From 1c6aabb3bbcc3e16e7d91140fd78e2a870e187cd Mon Sep 17 00:00:00 2001
From: Srujana Challa <schalla@marvell.com>
Date: Thu, 17 Dec 2020 15:15:47 +0530
Subject: [PATCH 684/767] octeontx2-af: Correct return value of
 ATTACH_RESOURCES mbox

commit f863eb873b650fc00aa9b883cfc32527b14492ab from
git@git.assembla.com:cavium/WindRiver.linux.git

If requested LFs are not available, return -ENOSPC instead of -EINVAL
from ATTACH_RESOURCES mailbox, So that user can distinguish
the error case.

Change-Id: I26465f187236d9a79db066c363c5306c466f3134
Signed-off-by: Srujana Challa <schalla@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/42516
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
index 34347dac106c..405eedf69c30 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
@@ -1571,11 +1571,13 @@ static int rvu_check_rsrc_availability(struct rvu *rvu,
 	int free_lfs, mappedlfs, blkaddr;
 	struct rvu_hwinfo *hw = rvu->hw;
 	struct rvu_block *block;
+	int ret;
 
-	if (rvu_check_rsrc_policy(rvu, req, pcifunc)) {
+	ret = rvu_check_rsrc_policy(rvu, req, pcifunc);
+	if (ret) {
 		dev_err(rvu->dev, "Func 0x%x: Resource policy check failed\n",
 			pcifunc);
-		return -EINVAL;
+		return ret;
 	}
 
 	/* Only one NPA LF can be attached */
-- 
2.31.1

